### PR TITLE
Factor LibraryImports into Common\ for System.Diagnostics.EventLog

### PR DESCRIPTION
**Pull request by deeprobin**
Factor LibraryImports into Common\ - See #28035

### CODE REVIEW

File: src/libraries/System.Diagnostics.EventLog/System.Diagnostics.EventLog.sln [hunk 1]
```diff
@@ -1,4 +1,8 @@
-﻿Microsoft Visual Studio Solution File, Format Version 12.00
```
Comment (deeprobin) [inline src/libraries/System.Diagnostics.EventLog/System.Diagnostics.EventLog.sln:1]: TODO @me: remove changes

**Comment by AaronRobinsonMSFT** [comment]
I'm not so sure this is better actually. All of these are for a very narrow scenario. I think I prefer them to be in a single file rather than spread across multiple. I realize it is against the recommendation, but these are all niche enough being in the same file just seems better.

/cc @dotnet/dotnet-diag

**Comment by noahfalk** [comment]
Including @dotnet/area-system-diagnostics-eventlog

**Comment by deeprobin** [comment]
> I'm not so sure this is better actually. All of these are for a very narrow scenario. I think I prefer them to be in a single file rather than spread across multiple. I realize it is against the recommendation, but these are all niche enough being in the same file just seems better.
> 
> /cc @dotnet/dotnet-diag
/cc @dotnet/area-system-diagnostics-eventlog

Hm, that's a design question.
If everything were in one file here, the file would of course be very large again. 

I have no idea if there are any other projects in the runtime repo that could use the wevtapi, but I think the primary consumer remains System.Diagnostics.EventLog.

Personally, I find the approach with the individual files better and more structured, but that's probably a subjective thing.

---

Another question is how to proceed with the EventLogHandle (internal). Are handles something that belongs in the Common\ folder or not?

**Comment by deeprobin** [comment]
Hey @ericstj, I think this is your area. Could you review the PR :smile:?

**Comment by ericstj** [comment]
> Hey @ericstj, I think this is your area. Could you review the PR 😄?

https://github.com/orgs/dotnet/teams/area-system-diagnostics-eventlog - @ViktorHofer @carlossanlop can you chime in.

I tend to share @noahfalk's opinion here.  I don't like churning all this code without a good reason.  If you really needed to share this code then that might be a good reason.  If not then it's just risk in a library that's legacy.  We do document a higher bar for this library: https://github.com/dotnet/runtime/blob/main/src/libraries/System.Diagnostics.EventLog/README.md#contribution-bar

**Comment by deeprobin** [comment]
@ericstj 
Thank you for your comments. This probably also applies to other libraries that could be labeled as “legacy” here.
I have put this in the issue - we can close it imo.

Why is System.Diagnostics.EventLog actually legacy? Even though I am a fan of ASP.NET applications on Linux instances, there are cases where applications are deployed on Windows. 
Sure, you could write to a log file here, but what's wrong with a central log component from the operating system. For Linux the equivalent would be syslog I think.

**Comment by ericstj** [comment]
I believe this is considered legacy because newer diagnostics tech (System.Diagnostics.DiagnosticsSource / OTel / etc) is recommended.

**Pull request closed without merging**